dipy/align/imaffine.py

```diff
@@ -0,0 +1,777 @@
+    r""" Transformation to align the center of mass of the input images
+
+    Parameters
+    ----------
+    static: array, shape(S, R, C)
+        static image
+    static_grid2space: array, shape (4, 4)
+        the voxel-to-space transformation of the static image
+    moving: array, shape(S, R, C)
+        moving image
+    moving_grid2space: array, shape (4, 4)
+        the voxel-to-space transformation of the moving image
+
+    Returns
+    -------
+    transform : array, shape(4, 4)
+        the affine transformation (translation only, in this case) aligning
+        the center of mass of the moving image towards the one of the static
+        image
+    """
+    dim = len(static.shape)
+    if static_grid2space is None:
+        static_grid2space = np.eye(dim + 1)
+    if moving_grid2space is None:
+        moving_grid2space = np.eye(dim + 1)
+    c_static = ndimage.measurements.center_of_mass(np.array(static))
+    c_static = static_grid2space.dot(c_static+(1,))
+    c_moving = ndimage.measurements.center_of_mass(np.array(moving))
+    c_moving = moving_grid2space.dot(c_moving+(1,))
+    transform = np.eye(dim + 1)
+    transform[:dim, dim] = (c_moving - c_static)[:dim]
+    return transform
+
+
+def aff_geometric_centers(static, static_grid2space, moving,
+                          moving_grid2space):
+    r""" Transformation to align the geometric center of the input images
+
+    With "geometric center" of a volume we mean the physical coordinates of
+    its central voxel
+
+    Parameters
+    ----------
+    static: array, shape(S, R, C)
+        static image
+    static_grid2space: array, shape (4, 4)
+        the voxel-to-space transformation of the static image
+    moving: array, shape(S, R, C)
+        moving image
+    moving_grid2space: array, shape (4, 4)
+        the voxel-to-space transformation of the moving image
+
+    Returns
+    -------
+    transform : array, shape(4, 4)
+        the affine transformation (translation only, in this case) aligning
+        the geometric center of the moving image towards the one of the static
+        image
+    """
+    dim = len(static.shape)
+    if static_grid2space is None:
+        static_grid2space = np.eye(dim + 1)
+    if moving_grid2space is None:
+        moving_grid2space = np.eye(dim + 1)
+    c_static = tuple((np.array(static.shape, dtype=np.float64)) * 0.5)
+    c_static = static_grid2space.dot(c_static+(1,))
+    c_moving = tuple((np.array(moving.shape, dtype=np.float64)) * 0.5)
+    c_moving = moving_grid2space.dot(c_moving+(1,))
+    transform = np.eye(dim + 1)
+    transform[:dim, dim] = (c_moving - c_static)[:dim]
+    return transform
+
+
+def aff_origins(static, static_grid2space, moving, moving_grid2space):
+    r""" Transformation to align the origins of the input images
+
+    With "origin" of a volume we mean the physical coordinates of
+    voxel (0,0,0)
+
+    Parameters
+    ----------
+    static: array, shape(S, R, C)
```

Comment:
While we're at it, sphinx is rather finnicky, so it needs to be: 

    static : 

Rather than: 

    static:

(space after the colon)